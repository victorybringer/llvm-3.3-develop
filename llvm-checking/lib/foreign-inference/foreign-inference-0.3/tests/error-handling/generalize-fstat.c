#include <sys/types.h>
#include <sys/stat.h>
#include <unistd.h>

void reportError(void* p) {
}

int learnFrom(int fd) {
  struct stat s;
  int b = fstat(fd, &s);
  if(b < 0) {
    reportError(&s);
    return -22;
  }

  return b + 1;
}

int target(int foo) {
  if(foo == 90) {
    reportError(NULL);
    return -50;
  }

  return foo + 12;
}
