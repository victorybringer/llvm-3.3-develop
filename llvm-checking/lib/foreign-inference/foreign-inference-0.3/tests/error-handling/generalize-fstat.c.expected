fromList [ ("learnFrom", (fromList [ "reportError" ], ReturnConstantInt (fromList [-22])))
         , ("target", (fromList [ "reportError" ], ReturnConstantInt (fromList [-50])))
         ]